is-valid-turn.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+const isValidMove = require("./is-valid-move");
+const makeMove = require("./make-move");
+const _ = require("lodash");
+
+function isValidTurn(gameState, diceRoll, proposedMoves) {
+    function getAceyDeucey() {
+        const aceyDeucey = {};
+        diceRoll.forEach(function(roll, index) {
+            if (roll === 1 && !aceyDeucey.hasOwnProperty("oneLocation")) {
+                aceyDeucey.oneLocation = index; 
+            } else if (roll === 2 && !aceyDeucey.hasOwnProperty("twoLocation")) {
+                aceyDeucey.twoLocation = index;
+            } else if (!aceyDeucey.hasOwnProperty("doublesLocation")) {
+                aceyDeucey.doublesLocation = index;
+            } else {
+                aceyDeucey.isAceyDeucey = false;
+            }
+        });
+        return aceyDeucey;
+    }
+    
+    function proposedMovesMatchDice() {
+        if (diceRoll.length > 2) {
+                      
+            const aceyDeucey = getAceyDeucey();
+            if (aceyDeucey.isAceyDeucey === false) {
+                return false;
+            }
+
+            if (
+                !(proposedMoves[0].numberOfSpaces === 1 && proposedMoves[1].numberOfSpaces === 2) &&
+                !(proposedMoves[0].numberOfSpaces === 2 && proposedMoves[1].numberOfSpaces === 1)
+            ) {
+                return false;
+            }
+            
+            const sortedAceyDeuceyRoll = _.sortBy(diceRoll);
+            for (let i = 0; i < 3; i++) {
+                sortedAceyDeuceyRoll.push(diceRoll[aceyDeucey.doublesLocation]);
```

Comment:
If all you're using `doublesLocation` for is finding the doubles value, what if you just returned that directly? Maybe you just need a `getDoublesOfDiceRoll()` function that tells you what the doubled value is? Then this could be:

```js
const doubledRoll = getDoublesOfDiceRoll();
for (let i = 0; i < 3; i++) {
    sortedAceyDeucyRoll.push(doubledRoll);
}
```